projects/aca-content/src/lib/components/search/search-save/directive/save-search.directive.spec.ts

```diff
@@ -0,0 +1,88 @@
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatDialog } from '@angular/material/dialog';
+import { By } from '@angular/platform-browser';
+import { of } from 'rxjs';
+import { SaveSearchDirective } from './save-search.directive';
+import { SaveSearchDialogComponent } from '../dialog/save-search-dialog.component';
+
+@Component({
+  template: '<div [acaSaveSearch]="searchQuery" acaSaveSearchQuery="encodedQuery"></div>'
+})
+class TestComponent {
+  searchQuery = 'encodedQuery';
+}
+
+describe('SaveSearchDirective', () => {
+  let fixture: ComponentFixture<TestComponent>;
+  let element;
+  let dialog: MatDialog;
+  let dialogRefMock;
+
+  const event = {
+    type: 'click',
+    preventDefault: jasmine.createSpy('preventDefault')
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [TestComponent],
+      imports: [SaveSearchDirective],
+      providers: [{ provide: MatDialog, useValue: { open: () => {} } }]
+    });
+
+    fixture = TestBed.createComponent(TestComponent);
+    element = fixture.debugElement.query(By.directive(SaveSearchDirective));
+    dialog = TestBed.inject(MatDialog);
+    spyOn(dialog, 'open').and.returnValue(dialogRefMock);
+
+    dialogRefMock = {
+      afterClosed: jasmine.createSpy('afterClosed').and.returnValue(of(null))
+    };
+    fixture.detectChanges();
+  });
+
+  it('should prevent the default click action', () => {
+    element.triggerEventHandler('click', event);
+    expect(event.preventDefault).toHaveBeenCalled();
+  });
+
+  it('should open the dialog with the correct configuration', () => {
+    element.triggerEventHandler('click', event);
+
+    const expectedConfig = {
+      data: { searchUrl: 'encodedQuery' }
+    };
+
+    expect(dialog.open).toHaveBeenCalledWith(SaveSearchDialogComponent, expectedConfig);
+  });
+
+  it('should call dialog open when clicked', () => {
```

Comment:
The previous test is already checking if dialog.open is being called, so this test seems unneccesary